src/main/blackbox/blackbox.c

```diff
@@ -1675,10 +1671,25 @@ void blackboxUpdate(timeUs_t currentTimeUs)
  */
 void blackboxInit(void)
 {
+    blackboxResetIterationTimers();
+
+    // an I-frame is written every 32ms
+    // gyro.targetLooptime is 1000 for 1kHz loop, 500 for 2kHz loop etc, gyro.targetLooptime is rounded for short looptimes
+    if (gyro.targetLooptime == 31) { // rounded from 31.25us
```

Comment:
`switch(gyro.targetLooptime)` is easier to read

This special-casing may be confusing, generated blackboxIInterval will be non-monotonous. It may be better to force power-of-2 value for targetLooptime < 128 (using few `if`s) or targetLooptime < 1000 (rounding to nearest power of two)